lightning/src/ln/onchain_utils.rs

```diff
@@ -0,0 +1,892 @@
+		self.counterparty_delayed_payment_base_key.write(writer)?;
+		self.counterparty_htlc_base_key.write(writer)?;
+		self.preimage.write(writer)?;
+		self.htlc.write(writer)?;
+		Ok(())
+	}
+}
+
+impl Readable for CounterpartyHTLCOutput {
+	fn read<R: ::std::io::Read>(reader: &mut R) -> Result<Self, DecodeError> {
+		let per_commitment_point = Readable::read(reader)?;
+		let counterparty_delayed_payment_base_key = Readable::read(reader)?;
+		let counterparty_htlc_base_key = Readable::read(reader)?;
+		let preimage = Readable::read(reader)?;
+		let htlc = Readable::read(reader)?;
+		Ok(CounterpartyHTLCOutput {
+			per_commitment_point,
+			counterparty_delayed_payment_base_key,
+			counterparty_htlc_base_key,
+			preimage,
+			htlc
+		})
+	}
+}
+
+/// A struct to describe a holder htlc output, amount and preimage to generate a signature and
+/// solving witness. It is used by OnchainTxHandler to finalize a HTLC transaction claiming this
+/// output.
+#[derive(Clone, PartialEq)]
+pub(crate) struct HolderHTLCOutput {
+	preimage: Option<PaymentPreimage>,
+	amount: u64,
+}
+
+impl Writeable for HolderHTLCOutput {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), ::std::io::Error> {
+		self.preimage.write(writer)?;
+		writer.write_all(&byte_utils::be64_to_array(self.amount))?;
+		Ok(())
+	}
+}
+
+impl Readable for HolderHTLCOutput {
+	fn read<R: ::std::io::Read>(reader: &mut R) -> Result<Self, DecodeError> {
+		let preimage = Readable::read(reader)?;
+		let amount = Readable::read(reader)?;
+		Ok(HolderHTLCOutput {
+			preimage,
+			amount,
+		})
+	}
+}
+
+/// A struct to describe a holder funding output with the static witnessScript to claim it. It is
+/// used by OnchainTxHandler to finalize a holder commitment transaction claiming this output.
+#[derive(Clone, PartialEq)]
+pub(crate) struct HolderFundingOutput {
+	funding_redeemscript: Script,
+}
+
+impl Writeable for HolderFundingOutput {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), ::std::io::Error> {
+		self.funding_redeemscript.write(writer)?;
+		Ok(())
+	}
+}
+
+impl Readable for HolderFundingOutput {
+	fn read<R: ::std::io::Read>(reader: &mut R) -> Result<Self, DecodeError> {
+		Ok(HolderFundingOutput {
+			funding_redeemscript: Readable::read(reader)?,
+		})
+	}
+}
+
+/// An enum to describe a claim content which is generated by ChannelMonitor and
+/// used by OnchainTxHandler to regenerate feerate-bump transactions to settle claims.
+///
+/// Template may be either malleable (a justice tx, a counterparty HTLC tx) or lockdown (a holder htlc
+/// tx, a holder commitment tx, a pre-signed justice tx). Bumping can be a Replace-by-Fee, that way
+/// the claim-settlement tx in itself has its feerate increased or Child-Pay-For-Parent, a child
+/// of the claim tx has its feerate increased. For the latter case, access to the whole package
+/// sizea and pre-committed fee is required to compute an efficient bump.
+#[derive(Clone, PartialEq)]
+pub(crate) enum PackageTemplate {
```

Comment:
Good point, I think we can aggregate MalleableJusticeTx/CounterpartyHTLCTx on a lot of cases. Though I'm holding this on the discussion to switch to dynamic trait objects instead of an enum pseudo-interface.